api/files/views.py

```diff
@@ -84,12 +85,13 @@ class FileDetail(JSONAPIBaseView, generics.RetrieveUpdateAPIView, FileMixin):
 
     def get_serializer_class(self):
         try:
-            target = self.get_target()
+            file_node = self.get_file()
```

Comment:
I'm not sure. Using `get_file` here has more permission checking and it makes it easy to checking if the quickfiles serializer is necessary. I'm not sure why this was originally `get_target` though, so it could be problematic.